python-lib/ontology_tagging/formatter_by_document.py

```diff
@@ -0,0 +1,442 @@
+        Append the tag columns to the output dataframe self.output_df
+        If one_row_per_doc_json is True, new columns are:
+        - 'tag_json_categories' : dictionary of category (keys) and tags (value)
+        - 'tag_json_full': dictionary of category (keys).
+            Each category has for value a dictionary of tags and their associated datas (keywords, sentences, occurences of the tag)
+        If one_row_per_doc_json is False, new columns are :
+            - a column with lists of keywords
+            - one column by category, with lists of tags
+            - a column with concatenated matched sentences
+
+        Args:
+            row: pandas.Series , document from text_df
+            one_row_per_doc_json: Bool to know if the format is nested JSON
+            language_column: if not None, matcher will apply with the given language of the row
+
+        """
+        language = self._get_document_language(row, language_column)
+        document_to_match = self._get_document_to_match(row, language)
+        matched_sentence, keyword_list = [], []
+        tag_columns_for_json, line, line_full = (
+            defaultdict(),
+            defaultdict(list),
+            defaultdict(defaultdict),
+        )
+        for idx, sentence in enumerate(document_to_match):
+            original_sentence = row[self.text_column_tokenized]
+            for keyword in sentence.ents:
+                line, line_full = self._get_tags_in_row_category(
+                    match=keyword,
+                    line=line,
+                    line_full=line_full,
+                    sentence=original_sentence[idx],
+                    language=language,
+                )
+                keyword_list.append(keyword.text + " ")
+                matched_sentence.append(original_sentence[idx] + "\n")
+            tag_columns_for_json["tag_json_categories"] = self._fill_tags(
+                condition=(line and one_row_per_doc_json), value=dict(line)
+            )
+            tag_columns_for_json["tag_json_full"] = self._fill_tags(
+                condition=(line and one_row_per_doc_json),
+                value={
+                    column_name: dict(value) for column_name, value in line_full.items()
+                },
+            )
+        line = {column: self._fill_tags(True, line[column]) for column in line}
+        if keyword_list:
+            self.tag_keywords.append(self._fill_tags(True, unique_list(keyword_list)))
+        else:
+            self.tag_keywords.append(np.nan)
+        self.tag_sentences.append(" ".join(unique_list(matched_sentence)))
+        self.output_df = (
+            self.output_df.append(tag_columns_for_json, ignore_index=True)
+            if one_row_per_doc_json
+            else self.output_df.append(
+                line,
+                ignore_index=True,
+            )
+        )
+
+    def _get_tags_in_row_category(
```

Comment:
`_get_tags_in_sentence_category` ?

Reply:
This is outdated since I have removed this method in PR #21